File: backend/models/analysis.go
package models

import "time"

// DocumentAnalysis is the complete intelligence report for a document
type DocumentAnalysis struct {
	ID        string    `json:"id"`
	Filename  string    `json:"filename"`
	CreatedAt time.Time `json:"createdAt"`
	Summary   Summary   `json:"summary"`
	Entities  []Entity  `json:"entities"`
	RiskFlags []Flag    `json:"riskFlags"`
	Tone      string    `json:"tone"`
	RawText   string    `json:"rawText,omitempty"`
}

// Entity represents a named entity extracted from the document
type Entity struct {
	Text        string  `json:"text"`
	Type        string  `json:"type"` // PERSON | ORG | LOCATION | DATE | EVENT
	Confidence  float64 `json:"confidence"`
	Occurrences []int   `json:"occurrences"` // char offsets in source doc
}

// Summary contains the structured document summary
type Summary struct {
	Headline  string   `json:"headline"`
	KeyPoints []string `json:"keyPoints"`
	WordCount int      `json:"wordCount"`
}

// Flag represents a risk or significance flag
type Flag struct {
	Text     string `json:"text"`
	Reason   string `json:"reason"`
	Severity string `json:"severity"` // LOW | MEDIUM | HIGH
}

// DocumentRecord stores raw document + metadata
type DocumentRecord struct {
	ID       string `json:"id"`
	Filename string `json:"filename"`
	Text     string `json:"text"`
	MimeType string `json:"mimeType"`
}

File: frontend/types/analysis.ts
export interface DocumentAnalysis {
  id: string;
  filename: string;
  createdAt: string;
  summary: Summary;
  entities: Entity[];
  riskFlags: Flag[];
  tone: string;
  rawText?: string;
}

export interface Entity {
  text: string;
  type: 'PERSON' | 'ORG' | 'LOCATION' | 'DATE' | 'EVENT';
  confidence: number;
  occurrences: number[];
}

export interface Summary {
  headline: string;
  keyPoints: string[];
  wordCount: number;
}

export interface Flag {
  text: string;
  reason: string;
  severity: 'LOW' | 'MEDIUM' | 'HIGH';
}

export interface UploadResponse {
  id: string;
  filename: string;
  wordCount: number;
  createdAt: string;
}

export interface DocumentRecord {
  id: string;
  filename: string;
  text: string;
  mimeType: string;
}

export type EntityType = Entity['type'];
export type Severity = Flag['severity'];

export const ENTITY_COLORS: Record<EntityType, string> = {
  PERSON: '#60A5FA',    // blue
  ORG: '#A78BFA',       // purple
  LOCATION: '#34D399',  // green
  DATE: '#FBBF24',      // amber
  EVENT: '#F472B6',     // pink
};

export const SEVERITY_COLORS: Record<Severity, string> = {
  LOW: '#FCD34D',
  MEDIUM: '#F97316',
  HIGH: '#EF4444',
};
